src/provideEndpoints.ts

```diff
@@ -0,0 +1,69 @@
+import { Endpoints } from './models/core/Endpoints';
+import { Environment } from './models/core/Environment';
+import { CloudRegion } from './models/core/CloudRegion';
+import { ServingConfig } from './models/core/SearchConfig';
+
+export const defaultApiVersion = 20220511;
+
+/**
+ * Provides methods for getting various endpoints.
+ *
+ * @internal
+ */
+export class EndpointsFactory {
+  private readonly environment: Environment;
+  private readonly cloudRegion: CloudRegion;
+
+  constructor(config?: ServingConfig) {
+    this.environment = config?.environment || Environment.PROD;
+    this.cloudRegion = config?.cloudRegion || CloudRegion.US;
+  }
+
+  /** Provides the domain based on environment and cloud region.
+   * @internal
+   */
+  getDomain() {
+    return `https://${this.environment}-cdn.${this.cloudRegion}.yextapis.com`;
+  }
+
+  /**
+   * Provides all endpoints based on environment and cloud region.
+   * @internal
+   */
+  getEndpoints() {
+    return {
+      universalSearch: `${this.getDomain()}/v2/accounts/me/search/query`,
+      verticalSearch: `${this.getDomain()}/v2/accounts/me/search/vertical/query`,
+      questionSubmission: `${this.getDomain()}/v2/accounts/me/createQuestion`,
+      status: 'https://answersstatus.pagescdn.com',
+      universalAutocomplete: `${this.getDomain()}/v2/accounts/me/search/autocomplete`,
+      verticalAutocomplete: `${this.getDomain()}/v2/accounts/me/search/vertical/autocomplete`,
+      filterSearch: `${this.getDomain()}/v2/accounts/me/search/filtersearch`,
+    };
+  }
+}
+
+/**
+ * The endpoints to use for sandbox experiences.
+ *
+ * @deprecated Use {@link provideEndpoints} instead
+ *
+ * @public
+ */
+export const SandboxEndpoints: Required<Endpoints> =
+  new EndpointsFactory({ environment: Environment.SANDBOX, cloudRegion: CloudRegion.US })
+    .getEndpoints();
+
+/**
+ * Provides all endpoints based on environment and cloud region.
+ *
+ * @remarks
+ * Returns an {@link Endpoints} instance.
+ *
+ * @param config - serving config to use for the domain, defaults to Prod and US if not provided.
+ *
+ * @public
+ */
+export function provideEndpoints(config?: ServingConfig): Required<Endpoints> {
```

Comment:
Since the environment and cloud region are part of the SearchConfig, I don't think we need this helper function anymore

Reply:
Don't we still need this if we don't want to expose EndpointsFactory class?

Reply:
Oh I understand what you mean. So they will always use provideCore. Sure. I'll remove this

Reply:
It looks like the function still exists?

Reply:
oops, forgot to include the file. Now actually removed!